File: BallClock_Scala/src/main/scala/BallClockMins.scala
/**
  * Created by joesturzenegger on 12/11/15.
  */

class BallClockMins(ballCount: Int, minuteCount: Int) extends BallClock(ballCount) {

  override def runClock(){
    for (i <- 1 to minuteCount){
      addMinute(mainQueue(0))
      mainQueue = mainQueue.drop(1)
    }
    println(getJSON)
  }
  def getJSON(): String = {
    val m = getString(mins)
    val f = getString(fiveMins)
    val h = getString(hours)
    val mq = getString(mainQueue)
    val s: String = "{" + "\"Min\": "+ m +"," + "\"FiveMin\": "+ f +"," +"\"Hour\": "+ h +"," +"\"Main\": " + mq + "}"
    s
  }
  def getString(a: List[Int]): String ={
    val s = a.mkString("[",",","]")
    s
  }


}

File: BallClock_Swift/BallClock/BallClockMins.swift
//
//  BallClockMins.swift
//  BallClock
//
//  Created by Joe Sturzenegger on 5/5/15.
//  Copyright (c) 2015 Joe Sturzenegger. All rights reserved.
//

import Foundation

class BallClockMins: BallClock
{
    var minuteCount: Int
    init(ballCount: Int, minuteCount: Int)
    {
        self.minuteCount = minuteCount
        super.init(ballCount: ballCount)
    }
    
    override func runClock()
    {
        for i in (1...minuteCount)
        {
            addMinute(mainQueue.removeAtIndex(0))
        }
        let jsonObjectHash = getJSON()
        print(jsonObjectHash)
    }
    func getJSON()->String
    {
        
        let min: String = "\"Min\":\(mins),"
        let five: String = "\"FiveMin\":\(fiveMins),"
        let hr: String = "\"Hour\":\(hours),"
        let main: String = "\"Main\":\(mainQueue)"
        
        return "{" + min + five + hr + main + "}"
    }
}